js-pkg/apps/demos/src/components/webrtc.js

```diff
@@ -0,0 +1,152 @@
+import * as React from 'react'
+import { useDriftDBSignalingChannel } from './signaling'
+
+export const useWebRTCMessagingChannel = (p1, p2) => {
+  let theirdataChannelRef = React.useRef(null)
+  let [messages, addMessage] = React.useReducer((state, msg) => [...state, msg], [])
+  let [latency, setLatency] = React.useState(0)
+  const connSetupArray = React.useRef([
+    (conn) => {
+      let dataChannel = conn.createDataChannel(p1)
+      dataChannel.onmessage = (e) => {
+        addMessage({ id: e.timeStamp, text: e.data })
+      }
+
+      conn.ondatachannel = (e) => {
+        if (e.channel.label === p2) {
+          theirdataChannelRef.current = e.channel
+        }
+      }
+    }
+  ])
+  const getLatency = useWebRTCConnection(p1, p2, connSetupArray.current)
+  React.useEffect(() => {
+    setTimeout(async function l() {
+      try {
+        const _lat = await getLatency()
+        setLatency(_lat)
+      } catch (e) {}
+    }, 1000)
+  }, [])
```

Comment:
See other comment about simplifying the expected behavior of `getLatency`, that way you can remove this try/catch.

Also, technically, we should probably cancel this timeout on unmount, otherwise we could end up calling `setLatency()` on a dead component, but since this is just a demo, maybe it's fine.

Reply:
done :+1: